app/src/lib/stores/app-store.ts

```diff
@@ -3347,44 +3348,98 @@ export class AppStore extends TypedBaseStore<IAppState> {
   public async _deleteBranch(
     repository: Repository,
     branch: Branch,
-    includeRemote: boolean
+    includeRemote?: boolean
   ): Promise<void> {
     return this.withAuthenticatingUser(repository, async (r, account) => {
-      const { branchesState } = this.repositoryStateCache.get(r)
-      let branchToCheckout = branchesState.defaultBranch
-
-      // If the default branch is null, use the most recent branch excluding the branch
-      // the branch to delete as the branch to checkout.
-      if (branchToCheckout === null) {
-        let i = 0
-
-        while (i < branchesState.recentBranches.length) {
-          if (branchesState.recentBranches[i].name !== branch.name) {
-            branchToCheckout = branchesState.recentBranches[i]
-            break
-          }
-          i++
-        }
+      const gitStore = this.gitStoreCache.get(r)
+
+      // If soley a remote branch, there is no need to checkout a branch.
+      if (branch.type === BranchType.Remote) {
+        await gitStore.performFailableOperation(() => {
+          // Note: deleting a remote branch implementation not needed, yet.
+          return Promise.resolve()
+        })
+
+        return this._refreshRepository(r)
```

Comment:
AFAIK none of this is needed:
- The `performFailableOperation` is only used to wrap `git` calls (which is not happening here).
- `_refreshRepository` should have nothing to refresh since we're not doing anything at all for remote branches.

Am I missing anything? 😳 
```suggestion
        return
```

Reply:
Oh, unless you're just leaving a "placeholder" for when the implementation for remote branches is added 😅  (I'd still prefer to remove it and add the implementation in the right PR, but I don't have a strong opinion about it since that new PR is almost done already 😆 ).

Reply:
Just leaving as placeholder - Since I am planning merging this into the add context menu one today. I would prefer to keep it.